test/test-lex.cpp

```diff
@@ -1017,6 +1017,11 @@ world`)"_sv,
       u8"`unterminated\\"_sv,  //
       u8"^^^^^^^^^^^^^^^ Diag_Unclosed_Template"_diag,
       {Token_Type::complete_template});
+
+  this->check_tokens_with_errors(
+      u8"`${}`"_sv,  //
+      u8"  ^^ Diag_Expected_Expression_In_Template_Literal"_diag,
```

Comment:
Nit: I haven't thought about this much, but perhaps it is better to put the span between the `{}`:
```suggestion
      u8"   ` Diag_Expected_Expression_In_Template_Literal"_diag,
```